cylc/rose/entry_points.py

```diff
@@ -75,6 +75,11 @@ def post_install(srcdir=None, opts=None, rundir=None):
     if results['fileinstall']:
         dump_rose_log(rundir=rundir, node=results['fileinstall'])
 
+    # Having dumped the config we clear rose options:
+    opts.rose_template_vars = []
+    opts.opt_conf_keys = []
+    opts.defines = []
```

Comment:
Pylance points out `opts` could be `None`.

However I don't think any of these arguments should have a default in the function definition
```diff
-def post_install(srcdir=None, opts=None, rundir=None):
+def post_install(srcdir, opts, rundir):
```

Reply:
The `post_install` interface hasn't always looked this way, probably hangover:

https://github.com/cylc/cylc-flow/blob/54959200e337c66df4896c066f042ffd2745f0d1/cylc/flow/scripts/install.py#L107-L109

We can drop the `=None` stuff (tests depending) with confidence because of Cylc version pinning. Just check the oldest cylc-flow version in range to make sure it's still compatible.